release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+  }
+
+  private String getFormattedReleaseNotes(List<ReleaseNoteText> releaseNotes) {
+    StringBuilder builder = new StringBuilder();
+    releaseNotes.forEach(
+        rnText -> {
+          builder.append(String.format("- %s (", rnText.text));
+          rnText.prNumbers.forEach(prNum -> builder.append(String.format("#%s ", prNum)));
+          builder.deleteCharAt(builder.length() - 1); // delete the last space character
+          builder.append(")\n");
+        });
+    return builder.toString();
+  }
+
+  private void checkReleaseNoteCategory(ReleaseNoteText rnText) {
+    if (rnText.category == null) {
+      rnText.category = Category.MISCELLANEOUS;
+    }
+  }
+
+  private boolean isContainedInSameAsItems(ReleaseNoteText rnText) {
+    return sameAsItems.values().stream().anyMatch(items -> items.contains(rnText));
+  }
+
+  public enum Category {
+    ENHANCEMENT("Enhancements"),
+    IMPROVEMENT("Improvements"),
+    BUGFIX("Bug fixes"),
+    DOCUMENTATION("Documentation"),
+    MISCELLANEOUS("Miscellaneous");
+
+    private final String displayName;
+
+    Category(String displayName) {
+      this.displayName = displayName;
+    }
+
+    public String getDisplayName() {
+      return this.displayName;
+    }
+
+    public static Category getByName(String name) {
+      return Arrays.stream(Category.values())
+          .filter(v -> v.name().equalsIgnoreCase(name))
+          .findFirst()
+          .orElseThrow(() -> new IllegalArgumentException("Invalid name: " + name));
+    }
+  }
+
+  static class ReleaseNoteText {
+    public Category category;
+    public String text;
+    public List<String> prNumbers = new ArrayList<>();
+  }
+
+  public static class GitHubContext {
+
+    private static final String MERGED_STATE = "merged";
+
+    private final String owner;
+    private final String projectTitlePrefix;
+    private final String version;
+    private final String repository;
+
+    public GitHubContext(String owner, String projectTitString, String version, String repository) {
+      this.owner = owner;
+      this.projectTitlePrefix = projectTitString;
+      this.version = version;
+      this.repository = repository;
+    }
+
+    public String getProjectId() throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh project list --owner %s | awk '/%s/ {print}' | awk '/%s/ {print $1}'",
+                  this.owner, this.projectTitlePrefix, this.version));
+
+      String line = br.readLine(); // Assuming only one line exists.
+      if (line == null) throw new RuntimeException("Couldn't get the projectId");
+      return line;
+    }
+
+    public List<String> getPullRequestNumbers(String projectId) throws Exception {
+      BufferedReader br =
+          runSubProcessAndGetOutputAsReader(
+              format(
+                  "gh project item-list %s --owner %s --limit 200 | awk -F'\\t' '/%s\\t/ {print"
```

Comment:
I would recommend using the `--json` option to make gh output in the json format and processing the result in Java (or using `jq`) because it would be helpful for readers to understand what output we need.

Reply:
I see. I'll revise it. Thank you.

Reply:
Sorry for flipping my state. I agree with you. But for now, I'd like to keep the code as is because it works without any problem for now. In addition, it is used only for creating release notes when we release the new version of Scalar products, not for production.

I will revise it according to your advice in another PR when I get a chance to revise it (e.g. Kaizen-day).